tests/test_flow/fakes.py

```diff
@@ -171,28 +117,65 @@ def pipe(self, url, content_bytes):
             f.write(content_bytes)
 
 
-def fake_aip_client(gcs_fs, caplog):
-    def create_aip_job(body, parent):
-        # AIP executions do not transmit logs to local instance of bionic.
-        # Emulate this behavior by setting the log level; this should filter out
-        # all (or almost all) the logs. Tests that check the log output will
-        # only test against logs which are printed locally.
-        with caplog.at_level(logging.CRITICAL):
-            assert body["trainingInput"]["args"][:3] == [
-                "python",
-                "-m",
-                "bionic.aip.main",
-            ]
-            run_aip(body["trainingInput"]["args"][3], gcs_fs)
-        return Mock()
+@contextmanager
+def temporarily_filter_all_logs():
+    class FilterAllLogs(logging.Filter):
+        def filter(self, rec):
+            return 0
+
+    logger = logging.getLogger()
+    filter_all_logs = FilterAllLogs()
+    try:
+        logger.addFilter(filter_all_logs)
+        yield
+    finally:
+        logger.removeFilter(filter_all_logs)
+
+
+@contextmanager
+def temporarily_clean_cache_dir(tmp_path):
```

Comment:
Is it possible for another process to be running that's accessing the same directory? For example, if we're using parallel execution?

Reply:
Good catch.

One idea I have is to make the parallel executor synchronous for tests.

Reply:
Switched to a different approach for file isolation for fake AIP. The new approach will extract the `TaskCompletionRunner` object from fake GCS, modify the cache directory path, and save it back into fake GCS.